### PR TITLE
Remove extra template params from sycl_iterator

### PR DESCRIPTION
**Pull request by SergeyKopienko**

### CODE REVIEW

**Comment by SergeyKopienko** [comment]
We have the next code in the file https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/internal/function.h :
```
// Helpers used to get indexable access to the data passed to the SYCL implementation of an
// algorithm from either a SYCL iterator or a USM pointer.
template <sycl::access::mode Mode, typename Policy, typename Iterator>
auto
get_access(Policy, Iterator i, typename ::std::enable_if<is_hetero_iterator<Iterator>::value, void>::type* = nullptr)
    -> decltype(i.get_buffer().template get_access<Mode>())
{
    return i.get_buffer().template get_access<Mode>();
}
```

I want to notice that in this place param ```Mode``` is template param and assess mode from the sycl_iterator class don't used here.

File: include/oneapi/dpl/internal/iterator_impl.h [hunk 1]
```diff
@@ -1,61 +0,0 @@
...
-    {
-        return i;
-    }
-};
-
-#if _ONEDPL_BACKEND_SYCL
-// Specialization for sycl_iterator to provide access to its component types needed to
-// construct the accessor type
-template <sycl::access::mode Mode, typename T, typename Allocator>
-struct extract_accessor<oneapi::dpl::__internal::sycl_iterator<Mode, T, Allocator>>
```
Comment (SergeyKopienko) [inline include/oneapi/dpl/internal/iterator_impl.h:43]: ```extract_accessor``` never used in our code.

File: include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h [hunk 2]
```diff
@@ -110,89 +104,81 @@ struct sycl_iterator
...
 {
-    return __internal::sycl_iterator<access_mode::read_write, T, Allocator>{buf, __dpl_sycl::__get_buffer_size(buf)};
+    return {buf, __dpl_sycl::__get_buffer_size(buf)};
 }
 
-// begin
-template <typename T, typename Allocator, access_mode Mode>
-__internal::sycl_iterator<Mode, T, Allocator> begin(sycl::buffer<T, /*dim=*/1, Allocator> buf, sycl::mode_tag_t<Mode>)
+// Old variants of begin/end for compatibility with old code
+inline namespace depreated
```
Comment (MikeDvorskiy) [inline include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h:132]: => `deprecated`

Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h]: Fixed

File: include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h [hunk 3]
```diff
@@ -110,89 +104,81 @@ struct sycl_iterator
...
+    using access_mode = sycl::access::mode;
 
-template <typename T, typename Allocator, access_mode Mode>
-__internal::sycl_iterator<__internal::_ModeConverter<Mode>::__value, T, Allocator>
+    // begin
+    template <typename T, typename Allocator, access_mode Mode>
+    __internal::sycl_iterator<T, Allocator>
+    begin(sycl::buffer<T, /*dim=*/1, Allocator> buf, sycl::mode_tag_t<Mode>)
+    {
+        return oneapi::dpl::begin(buf);
```
Comment (MikeDvorskiy) [inline include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h:141]: I think we have to print a kind of "deprecation" warning on each call of that deprecated  API.

Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h:142]: Fixed

**Pull request closed without merging**